tests/test_decorators.py

```diff
@@ -25,14 +28,14 @@ def test_func2(a, b: str, c=None, d: list = None):
 failures1 = [
     (1, 1, [], []),  # first annotation wrong
     (1, "hello", [], 1),  # second annotation wrong
-    (1, 1, 1, 1)  # both annoations wrong
+    (1, 1, 1, 1)  # both annotations wrong
 ]
 
-correct2 = {'a': 1, 'b': "hello", 'c': [], 'd': []}
+correct2 = OrderedDict(a=1, b="hello", c=[], d=[])
 failures2 = [
-    {'a': 1, 'b': 1, 'c': [], 'd': []},  # first annotation wrong
```

Comment:
Changed to OrderedDicts, as we rely on the ordering for the args. The ordering, although currently given in CPython, is not a feature but unreliable implementation detail, so OrderedDicts are the way to go